### PR TITLE
Added !dominos

### PR DESCRIPTION
**Pull request by ldgrp**
Reimplements `dominos.js`

### CODE REVIEW

**Comment by nicklambourne** [comment]
lgtm, can you test it in the uqcstesting slack and throw up some images?

**Comment by ldgrp** [comment]
screenshot 2018-09-13 22 36 05

**Comment by nicklambourne** [comment]
@TRManderson can you give this a quick onceover/lgtm (+ thoughts on list over List[] in typing) and I'll merge it?

**Comment by TRManderson** [comment]
Looks really good, but having some tests would be nice.

@ldgrp - check out https://docs.python.org/3/library/typing.html RE your type annotations.

`list` is effectively `List[Any]`

**Comment by ldgrp** [comment]
I'll try mimicking the tests for !umart if that's alright.

**Comment by ldgrp** [comment]
I'll try mimicking the tests for [!umart](https://github.com/UQComputingSociety/uqcsbot/blob/master/test/test_umart.py) if that's alright

File: uqcsbot/scripts/dominos.py [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+            any(coupon.keyword_matches(keyword) for keyword in keywords)]
+        
+def get_coupons(n: int, ignore_expiry: bool, keywords: List[str]) -> List[Coupon]:
+    '''
+    Returns a list of n Coupons
+    '''
+
+    coupon_page = get_coupon_page()
+    if coupon_page is None:
+        return None
```
Comment (MaxwellBo) [inline uqcsbot/scripts/dominos.py]: If this is returning `None`, shouldn't our type be annotated with `Optional[List[Coupon]]`?

**Pull request merged**